### PR TITLE
Optimize KVStore::iterate to prevent unecessary read of deleted entries

### LINKED ISSUE #3098: Be hang when starting
### Steps to reproduce the behavior (Required)
In our scenoria, we create some big primary key table and use flink continuously write data into it. when I restarting the be process, it hanged.
It has about 950mb meta file on the disk
image

### Expected behavior (Required)
Starting quickly

### Real behavior (Required)
image

the be process maybe hang in Rockdb::Seek the meta file and this function actually hang about 20min.

### StarRocks version (Required)
2.0.1

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #3098 

## Problem Summary(Required) ：

When BE starts, it will load all tablets; for the primary key tablet, the load process will call `KVStore::iterate` several times to load logs, pending_rowsets, rowsets and delvecs. They all call `rocksdb::Iterator::Seek` to do some prefix range query. Currently, its implementation is suboptimal; for example, suppose the records in RocksDB is:
```
'a' 'd0001'(deleted) 'd0002'(deleted) .. 'd9999'(deleted) 'e'
``` 
`Seek('c')` will first iterate all 9999 deleted entries, then seek to 'e', then find there is no record with prefix 'c', which is inefficient. 
This PR sets the iterate_upper_bound option when doing prefix range query, so for the above example, seek will stop at record 'd0001' and stop.

In my test with data provided by the user in #3098, before this PR, KVStore::iterate takes >100ms, and the whole load process takes >20min; after this PR, each call takes <1ms the entire load process only takes 3s.

### CODE REVIEW

File: be/src/storage/kv_store.cpp [hunk 1]
```diff
@@ -182,10 +182,30 @@ Status KVStore::remove(ColumnFamilyIndex column_family_index, const std::string&
     return to_status(s);
 }
 
+static std::string get_prefix_end(const std::string& prefix) {
+    std::string ret = prefix;
+    for (ssize_t i = (ssize_t)ret.size() - 1; i >= 0; i--) {
+        if ((uint8_t)ret[i] < 255) {
+            ret[i]++;
```
Comment (chaoyli) [inline be/src/storage/kv_store.cpp:189]: What's the purpose of plus ret[i]?

Comment (decster) [reply be/src/storage/kv_store.cpp:189]: for example:
```
prefix(ascii):      00001
prefix_end(ascii):  00002

prefix(hex):      00ffff
prefix_end(hex):  010000

```

**Comment by decster** [comment]
run starrocks_be_unittest

**Comment by decster** [comment]
run starrocks_be_unittest

**Review by chaoyli — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**